### ISSUE TITLE
Guests can start new conversations ad libitum

### ISSUE THREAD

**Issue by simonspa** [nextcloud/talk-android#1720]
I realized that despite having only a limited guest account, users can start any desired conversation (such as public ones, or one-to-one conversations with any user on the server) via the Android Talk app (13.0.0 RC1 via GPlay Beta)

## Steps to reproduce
1. Log into server with a guest account
2. Click on "+" to start new conversation, select what you would like to do
3. Create conversations ad libitum

### Expected behaviour
Guest accounts should have the same restrictions in the Android app as in the server app - but judging from the possibility to create any conversation desired, this is also linked to the Talk server API that doesn't prevent this?

### Actual behaviour
Guests can start any conversation the want, including one-to-one with full search capabilities among all server users.

## Device information

**Device:** Google Pixel 4
**Android version:** 12

**Talk version:** 13.0.0

## Server information

**Nextcloud version:** 23 RC

**Talk version:** 13

**Comment by mahibi** [comment]
i first was confused what you mean with "guest user" (you don't mean an invite without account, but just a "normal" user who is restricted for some actions by the talk settings, right?)

so now i reproduced and approved this issue.

- restrict "Limit creating a public and group conversation" in talk settings to only some group.

![grafik](https://user-images.githubusercontent.com/2932790/146909896-30e1d845-b8a5-4e56-962b-5cc2f5923899.png)

- login with a user who is not in this group via web -> not possible to create a new conversation
- login with a user who is not in this group via android talk -> it's possible to create a new conversation!

**Comment by simonspa** [comment]
I really mean "guest" as in "an account created via the Guests app" - however, I think by now the list of users I'm seeing is just the other guest accounts but not the actual users. Still, this might not be desired?

**Comment by mahibi** [comment]
yes a "guest" created via the Guests app i considered to be a "normal" user. so it doesn't matter if the guest app was used or the user was created by admins. The restriction to not create new conversations just fails!

**Comment by mahibi** [comment]
> however, I think by now the list of users I'm seeing is just the other guest accounts but not the actual users. Still, this might not be desired?

will check this in seperate
